core/templates/components/ck-editor-helpers/ck-editor-copy-content-service.ts

```diff
@@ -67,6 +67,7 @@ export class CkEditorCopyContentService {
       const currentTagName = currentElement.tagName.toLowerCase();
       if (currentTagName.includes(this.NON_INTERACTIVE_TAG)) {
         containedWidgetTagName = currentTagName;
+        break;
```

Comment:
We are breaking here because currently, the copy method returns the root element for all the cases and the paste method doesn't work as expected for the following situation:

Text: `Some text <math-expression> other text <math-expression-2>.`

If in the following text a user tried to copy the 1st math-expression then the 2nd math-expression gets copied.

Reply:
How would this `break` fix that? Aren't they 2 different elements and this loop just goes over the parents?
Also, why is this loop required? Since `containedWidgetTagName` is set in the next loop anyway? (or is that not reached)

Reply:
@aks681, The two-loops are for moving up and down the dom tree. This is needed as we wanted to copy the entire nearest noninteractive tag. (The click can happen on the element inside the noninteractive tag for e.g, the math expressions are images so when you click the math expressions you are clicking on img tag.)  

@sagangwee, This is an expected code for this function, I think it was missing earlier unexpectedly. This also makes it more readable i.e, we are breaking once we found the important noninteractive element/tag to copy! If required, I can add it in a seperate PR! (I want to get this merged

Reply:
Ah ok, makes sense.